src/evaluators/utility/PHAL_LogGaussianDistributedParameter.hpp

```diff
@@ -0,0 +1,324 @@
+
+namespace PHAL {
+///
+/// LogGaussianDistributedParameterBase
+///
+template<typename EvalT, typename Traits>
+class LogGaussianDistributedParameterBase : 
+  public PHX::EvaluatorWithBaseImpl<Traits>,
+  public PHX::EvaluatorDerived<EvalT, Traits>
+{
+ private:
+  typedef typename EvalT::ParamScalarT ParamScalarT;
+
+ public:
+  typedef typename EvalT::ScalarT   ScalarT;
+  //typedef ParamNameEnum             EnumType;
+
+  LogGaussianDistributedParameterBase (const Teuchos::ParameterList& p, const Teuchos::RCP<Albany::Layouts>& dl)
+  {
+    std::string log_gaussian_field_name   = p.get<std::string>("Log Gaussian Name");
+    std::string gaussian_field_name       = p.get<std::string>("Gaussian Name");
+    std::string mean_field_name;
+
+    mean_is_field = false;
+    if (p.isParameter("Mean Name")) {
+      mean_field_name = p.get<std::string>("Mean Name");
+      mean_is_field = true;
+    }
+
+    eval_on_side = false;
+    if (p.isParameter("Side Set Name")) {
+      sideSetName = p.get<std::string>("Side Set Name");
+      eval_on_side = true;
+    }
+
+    logGaussian = decltype(logGaussian)(log_gaussian_field_name,dl->node_scalar);
+    numNodes = 0;
+
+    gaussian = decltype(gaussian)(gaussian_field_name,dl->node_scalar);
+
+    if (mean_is_field) {
+      mean = decltype(mean)(mean_field_name,dl->node_scalar);
+      this->addDependentField(mean);
+    }
+    else {
+      RealType mean = p.get<RealType>("mean");
+      RealType deviation = p.get<RealType>("deviation");
+      a = log(mean/sqrt(1.+deviation*deviation));
+      b = sqrt(log(1+deviation*deviation));
+    }  
+    this->addEvaluatedField(logGaussian);
+    this->addDependentField(gaussian);
+  
+    this->setName("Log Gaussian " + log_gaussian_field_name + PHX::print<EvalT>());
+  }
+
+  void postRegistrationSetup(typename Traits::SetupData d, PHX::FieldManager<Traits>& fm)
+  {
+    this->utils.setFieldData(logGaussian,fm);
+    numNodes = logGaussian.extent(1);
+    d.fill_field_dependencies(this->dependentFields(),this->evaluatedFields(),false);
+  }
+
+protected:
+  std::size_t numNodes;
+  PHX::MDField<ScalarT,Cell,Node> logGaussian;
+  PHX::MDField<const ScalarT,Cell,Node> gaussian;
+  PHX::MDField<const RealType,Cell,Node> mean;
+  RealType a, b;
+
+  bool eval_on_side, mean_is_field;
+  std::string sideSetName;
+};
+
+template<typename EvalT, typename Traits> class LogGaussianDistributedParameter;
+
+// **************************************************************
+// **************************************************************
+// * Specializations
+// **************************************************************
+// **************************************************************
+
+
+// **************************************************************
+// Residual
+// **************************************************************
+template<typename Traits>
+class LogGaussianDistributedParameter<PHAL::AlbanyTraits::Residual,Traits>
```

Comment:
Like for lin comb: all specializations seem to do the same thing. Can we do everything in the base class?